MapboxNavigation/NavigationMapView.swift

```diff
@@ -125,49 +137,141 @@ open class NavigationMapView: MGLMapView {
+                } else {
+                    mergedCongestionSegments.append(seg)
+                }
+            }
+            
+            let lines = mergedCongestionSegments.map { (congestionSegment: CongestionSegment) -> MGLPolylineFeature in
+                let polyline = MGLPolylineFeature(coordinates: congestionSegment.0, count: UInt(congestionSegment.0.count))
+                polyline.attributes["congestion"] = String(describing: congestionSegment.1)
+                if let legIndex = legIndex {
+                    polyline.attributes[isOpaqueIdentifier] = index != legIndex
+                } else {
+                    polyline.attributes[isOpaqueIdentifier] = false
+                }
+                return polyline
+            }
+            
+            linesPerLeg.append(lines)
+        }
+        
+        return MGLShapeCollectionFeature(shapes: Array(linesPerLeg.joined()))
+    }
+    
+    func shape(describingCasing route: Route, legIndex: Int?) -> MGLShape? {
+        var linesPerLeg: [MGLPolylineFeature] = []
+        
+        for (index, leg) in route.legs.enumerated() {
+            let legCoordinates = Array(leg.steps.flatMap {
+                $0.coordinates
+            }.joined())
+            
+            let polyline = MGLPolylineFeature(coordinates: legCoordinates, count: UInt(legCoordinates.count))
+            if let legIndex = legIndex {
+                polyline.attributes[isOpaqueIdentifier] = index != legIndex
             } else {
-                mergedCongestionSegments.append(seg)
+                polyline.attributes[isOpaqueIdentifier] = false
             }
+            linesPerLeg.append(polyline)
         }
         
-        // Filter out any segments with low congestion
-        let nontrivialCongestionSegments = mergedCongestionSegments.filter { $0.1 != CongestionLevel.unknown && $0.1 != CongestionLevel.low }
+        return MGLShapeCollectionFeature(shapes: linesPerLeg)
+    }
+    
+    func shape(for waypoints: [Waypoint]) -> MGLShape? {
+        var features = [MGLPointFeature]()
+        let letters = (97...122).map({Character(UnicodeScalar($0))}).map { String(describing:$0).uppercased() }
         
-        let baseLine = MGLPolylineFeature(coordinates: coordinates, count: route.coordinateCount)
-        baseLine.attributes["congestion"] = "unknown"
-        let lines = nontrivialCongestionSegments.map { (congestionSegment: CongestionSegment) -> MGLPolylineFeature in
-            let polyline = MGLPolylineFeature(coordinates: congestionSegment.0, count: UInt(congestionSegment.0.count))
-            polyline.attributes["congestion"] = String(describing: congestionSegment.1)
-            return polyline
+        for (waypointIndex, waypoint) in waypoints.enumerated() {
+            let feature = MGLPointFeature()
+            feature.coordinate = waypoint.coordinate
+            feature.attributes = [ "name": letters[waypointIndex] ]
```

Comment:
If we’re going to label each waypoint with a letter, then the set of letters should be localizable, because each language has its own order. (For example, in Vietnamese it begins “A Ă Â B C D Đ E Ê …”. In Simplified Chinese, it begins with “万千百十…”.)

It would be great if we could use `Locale.exemplarCharacterSet`, but unfortunately there isn’t a way to enumerate a CharacterSet’s members while preserving order. We’ll need a single localizable string that contains a space- or comma-delimited list of exemplar characters.